link.go

```diff
@@ -840,9 +850,13 @@ func (l *link) muxDetach() {
 	// reattaching and then sending a closing detach."
 
 	l.detachErrorMu.Lock()
-	detachError := l.detachError
+	detachError, ok := l.detachError.(*Error)
```

Comment:
Ugh I didn't realize we needed to slurp this out.  How do you feel about this?  Personally, it makes me nervous, but maybe it's ok?

Reply:
I wasn't a big fan either. It's something that should "never" happen, but the now that it's an 'error' type it's not longer a simple compile error.

I've reverted back to the PR i started the entire thing with for now. We can have some further discussion around this because it's an annoying bug that I'd like to stomp out entirely for the future as well.